src/arrays/arraymath.jl

```diff
@@ -121,6 +121,11 @@ function \{B<:OrthonormalBasis}(op::AbstractQuMatrix{B}, vec::AbstractQuVector{B
     return QAT(div, bases(op,1))
 end
 
+# dot product of QuArray's
+dot{B<:OrthonormalBasis}(vec1::AbstractQuVector{B}, vec2::AbstractQuVector{B}) = dot(coeffs(vec1), coeffs(vec2))
```

Comment:
Hmm...I don't see why not. Currently `dot` throws an error if you try to input row vectors, but I suppose that's due more to the matrix/row vector type ambiguity rather than an actual API decision.